sdk/appconfiguration/azappconfig/client.go

```diff
@@ -0,0 +1,443 @@
+	}
+
+	return url.Scheme + "://" + url.Host + "/.default", nil
+}
+
+// NewClient returns a pointer to a Client object affinitized to an endpointUrl.
+func NewClient(endpointUrl string, cred azcore.TokenCredential, options *ClientOptions) (Client, error) {
+	if options == nil {
+		options = &ClientOptions{}
+	}
+
+	genOptions := options.toConnectionOptions()
+
+	tokenScope, err := getDefaultScope(endpointUrl)
+	if err != nil {
+		return Client{}, err
+	}
+
+	syncTokenPolicy := newSyncTokenPolicy()
+	genOptions.PerRetryPolicies = append(
+		genOptions.PerRetryPolicies,
+		runtime.NewBearerTokenPolicy(cred, []string{tokenScope}, nil),
+		syncTokenPolicy,
+	)
+
+	pl := runtime.NewPipeline(generated.ModuleName, generated.ModuleVersion, runtime.PipelineOptions{}, genOptions)
+	return Client{
+		appConfigClient: generated.NewAzureAppConfigurationClient(endpointUrl, nil, pl),
+		syncTokenPolicy: syncTokenPolicy,
+	}, nil
+}
+
+// NewClientFromConnectionString parses the connection string and returns a pointer to a Client object.
+func NewClientFromConnectionString(connectionString string, options *ClientOptions) (Client, error) {
+	if options == nil {
+		options = &ClientOptions{}
+	}
+
+	genOptions := options.toConnectionOptions()
+
+	endpoint, credential, secret, err := parseConnectionString(connectionString)
+	if err != nil {
+		return Client{}, err
+	}
+
+	syncTokenPolicy := newSyncTokenPolicy()
+	genOptions.PerRetryPolicies = append(
+		genOptions.PerRetryPolicies,
+		newHmacAuthenticationPolicy(credential, secret),
+		syncTokenPolicy,
+	)
+
+	pl := runtime.NewPipeline(generated.ModuleName, generated.ModuleVersion, runtime.PipelineOptions{}, genOptions)
+	return Client{
+		appConfigClient: generated.NewAzureAppConfigurationClient(endpoint, nil, pl),
+		syncTokenPolicy: syncTokenPolicy,
+	}, nil
+}
+
+// UpdateSyncToken sets an external synchronization token to ensure service requests receive up-to-date values.
+func (c *Client) UpdateSyncToken(token string) {
+	c.syncTokenPolicy.addToken(token)
+}
+
+func (cs Setting) toGeneratedPutOptions(ifMatch *azcore.ETag, ifNoneMatch *azcore.ETag) *generated.AzureAppConfigurationClientPutKeyValueOptions {
+	return &generated.AzureAppConfigurationClientPutKeyValueOptions{
+		Entity:      cs.toGenerated(),
+		IfMatch:     (*string)(ifMatch),
+		IfNoneMatch: (*string)(ifNoneMatch),
+		Label:       cs.Label,
+	}
+}
+
+// AddSettingResult contains the response from AddSetting method.
+type AddSettingResult struct {
+	Setting
+
+	// Sync token for the Azure App Configuration client, corresponding to the current state of the client.
+	SyncToken *string
+}
+
+func fromGeneratedAdd(g generated.AzureAppConfigurationClientPutKeyValueResponse) AddSettingResult {
+	return AddSettingResult{
+		Setting:   settingFromGenerated(g.KeyValue),
+		SyncToken: g.SyncToken,
+	}
+}
+
+// AddSettingOptions contains the optional parameters for the AddSetting method.
+type AddSettingOptions struct {
```

Comment:
Should have a comment like [this](https://github.com/Azure/autorest.go/blob/track2/test/autorest/complexgroup/zz_generated_response_types.go#L28)